draft-barnes-acme.md

```diff
@@ -1459,14 +1450,131 @@ TODO
+
+* Simple HTTP: The value provided in the validation request is signed by the account private key.
+* DVSNI: The validation TLS request uses the account key pair as the server's key pair.
+* DNS: The MAC key is derived from an ECDH key for which only the account key holder knows the private key.
+
+The "proof of possession" challenge, which requires no out-of-band query, is bound to an account key by having the prior key sign over the account public key, assuring that the signature cannot be replayed to authorize a different account key.
+
+The association of challenges to identifiers is typically done by requiring the client to respond to a validation query through a channel that only someon in control of the identifier can access.  Thus, each challenge makes an assumption about the underlying channel:
+
+* Simple HTTP: Only the domain owner can provision files under .well-known on a web server for the domain
+* DVSNI: Only the domain owner can configure a TLS server for the domain
+* DNS: Only the domain owner can provision records for the domain
+
+There are several ways that these assumptions can be violated, both by misconfiguration and by attack.  For example, on a web server that allows non-administrative users to write to .well-known, any user can claim to own the server's hostname by responding to a Simple HTTP challenge.
+
+The DNS is a common point of vulnerability for all of these challenges.  An entity that can provision false DNS records for a domain can attack the DNS challenge directly, and can provision false A/AAAA records to direct the ACME server to send its DVSNI or Simple HTTP validation query to a server of the attacker's choosing.  There are a few different mitigations that ACME servers can apply:
+
+* Checking the DNSSEC status of DNS records used in ACME validation
+* Querying the DNS from multiple vantage points to address local attackers
+* Performing DNS queries only over TCP, to require an on-path attacker
+
+Of course, if the owner of the domain has outsourced operation of the zone to a hosting provider, there is nothing that can be done against tampering by the hosting provider, since as far as the outside world is concerned, the zone provided by the hosting provider is the real zone.
+
+Given these considerations, the ACME validation process makes it impossible for any attacker on the ACME channel, or a passive attacker on the validation channel to hijack the authorization process to authorize a key of the attacker's choice.
+
+An attacker that can only see the ACME channel can't even generate a validation response that would satisfy the server, because of the binding to the account key.  A passive attacker on the validation channel can observe the correct validation response and even replay it, but that response can only be used with the account key for which it was generated.
```

Comment:
It's supposed to explain/justify the assertion in the previous para.  I have reworded to try to clarify.